src/operator/context/operator.tsx

```diff
@@ -0,0 +1,167 @@
+// Libraries
+import React, {FC, useCallback, useEffect, useState} from 'react'
+import {useDispatch} from 'react-redux'
+
+// Utils
+import {notify} from 'src/shared/actions/notifications'
+import {getMe, getAccounts, getOrgs} from 'src/operator/api'
+import {getAccountsError, getOrgsError} from 'src/shared/copy/notifications'
+
+// Types
+import {Account, Me, Organization} from 'src/types/operator'
+import {RemoteDataState} from 'src/types'
+
+export type Props = {
+  children: JSX.Element
+}
+
+export interface OperatorContextType {
+  accounts: Account[]
+  activeTab: string
+  handleGetAccounts: () => void
+  handleGetOrgs: () => void
+  operator: Me
+  operatorStatus: RemoteDataState
+  organizations: Organization[]
+  searchTerm: string
+  setActiveTab: (tab: string) => void
+  setSearchTerm: (searchTerm?: string) => void
+  status: RemoteDataState
+}
+
+export const DEFAULT_CONTEXT: OperatorContextType = {
+  accounts: [],
+  activeTab: 'organizations',
+  handleGetAccounts: () => {},
+  handleGetOrgs: () => {},
+  operator: null,
+  operatorStatus: RemoteDataState.NotStarted,
+  organizations: [],
+  searchTerm: '',
+  setActiveTab: () => {},
+  setSearchTerm: () => {},
+  status: RemoteDataState.NotStarted,
+}
+
+export const OperatorContext = React.createContext<OperatorContextType>(
+  DEFAULT_CONTEXT
+)
+
+export const OperatorProvider: FC<Props> = React.memo(({children}) => {
+  const [activeTab, setActiveTab] = useState('accounts')
+  const [accounts, setAccounts] = useState([])
+  const [operator, setOperator] = useState(null)
+  const [accountStatus, setAccountStatus] = useState(RemoteDataState.NotStarted)
+  const [orgsStatus, setOrgsStatus] = useState(RemoteDataState.NotStarted)
+  const [operatorStatus, setOperatorStatus] = useState(
+    RemoteDataState.NotStarted
+  )
+  const [searchTerm, setSearchTerm] = useState('')
+  const dispatch = useDispatch()
+
+  const [organizations, setOrganizations] = useState([])
+
+  // TODO(ariel): might need to debounce
```

Comment:
Yeah, I think you will. I think we fire off a new request on every character typed in the search bar, which is excessive if someone is typing reasonably fast.

Reply:
totally, I wanted to stay away from lodash but I'll add it all back in for now